### DISCUSSION TITLE
How can I add style for HTML tag?

### DISCUSSION THREAD

**Discussion by whale2002** [vercel/next.js#43372] [Help] [answered]
I want to add style for html dynamically through css-module, and I do not want to handle DOM in useEffect.  What can I do？

**Comment by icyJoseph** [comment, marked as answer]
Perhaps the simplest is to respond to user interactions of async work resolving/rejecting?

> You may not see the CSS changes because of specificity and cascading though.

Assuming, this CSS Module:

```css
.green {
  color: green;
}

.blue {
  color: blue;
  font-size: 3rem;
}

```

Then...

```tsx
// _app.tsx
import type { AppProps } from "next/app";

import style from "../styles/Html.module.css";

const applyToRoot = (className: string) => {
  const current = Array.from(document.documentElement.classList);
  document.documentElement.classList.remove(...current);

  document.documentElement.classList.add(className);
};

export default function App({ Component, pageProps }: AppProps) {
  return (
    <>
      <button onClick={() => applyToRoot(style.green)}>Make green</button>
      <button onClick={() => applyToRoot(style.blue)}>Make blue</button>

      <Component {...pageProps} />
    </>
  );
}
```

I guess the real question is, what do you mean with: 

>  and I do not want to handle DOM in useEffect

**Reply by whale2002** [reply]
In fact, I do this in SSR. And I hope this heppen in server.

**Reply by icyJoseph** [reply]
Aha that changes everything.

Ok, so you might want to do this in `_document` with a few considerations...

- Define the CSS modules classes as global, 
- Import them in `_app`, then apply the class name strings you need, conditionally in `_document`,
- Next.js tries to pre-render by default, so if you are not careful, you could pre-render with one class applied only
- `_document` is fairly static, but by explicitly fighting the pre-render by default, you can make it on-demand


Another way more CSS oriented would be to use `:root {}`, which is a CSS way of targeting the root of a document:

> The :root [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) matches the root element of a tree representing the document. In HTML, :root represents the [<html>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/html) element and is identical to the selector html, except that its [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) is higher.

My follow up would be, what does the dynamic part imply? A brand for a white-label application?

I feel like using some style tags and `:root` trickery you could achieve this during SSR, but what would the dynamic part depend on? Not sure if CSS modules would still apply in this case though 🤔
